hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/StreamReadMonitoringFunction.java

```diff
@@ -214,8 +215,10 @@ public void monitorDirAndForwardSplits(SourceContext<MergeOnReadInputSplit> cont
     }
     IncrementalInputSplits.Result result =
         incrementalInputSplits.inputSplits(metaClient, this.issuedOffset, this.cdcEnabled);
-    if (result.isEmpty()) {
+
+    if (result.isEmpty() && StringUtils.isNullOrEmpty(result.getEndInstant())) {
       // no new instants, returns early
+      LOG.info("result is empty, do not update issuedInstant.");
```

Comment:
result -> Result，and use log.warn instead, maybe we just eliminate the log message because it's almost useless.